### PR TITLE
Replace custom tooltips with native one for files sharing

### LINKED ISSUE #35654: [BITV]: Replace custom tooltips with native one for files_sharing
Replace custom tooltips with native one in view.

- [ ] Needs NC Vue release

## ``@nextcloud/vue``
- [x] https://github.com/nextcloud/nextcloud-vue/issues/3587
- [x] https://github.com/nextcloud/server/issues/35857
- [x] https://github.com/nextcloud/server/issues/35858
- [x] https://github.com/nextcloud/server/issues/35860
- [x] https://github.com/nextcloud/server/issues/35862

## Server

 - [ ]  SharingEntry.vue
 - [ ] SharingEntryInherited.vue
 - [ ] SharingEntryInternal.vue
 - [ ] SharingEntryLink.vue
 - [ ] SharingEntrySimple.vue
 - [ ] SharingInherited.vue
 - [ ] SharingTab.vue
 - [ ] SharesMixin.js
 - [ ] files_drop.js
 - [ ] public.js
 - [ ] sharedfilelist.js
 - [ ] share.js

This is a part of https://github.com/nextcloud/nextcloud-vue/issues/2503

### PR DESCRIPTION
**Pull request by JuliaKirschenheuter**
**Resolves only js/jQuery part of: https://github.com/nextcloud/server/issues/35654**

Fixes https://github.com/nextcloud/server/issues/35654

## Summary

- [ ]  Replace custom tooltips with native ones in js files for files_sharing
## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by JuliaKirschenheuter** [comment]
I was not able to test: 

https://localhost/apps/files/?dir=/&view=deletedshares
https://localhost/apps/files/?dir=/&view=pendingshares

Reviewers: please help me with it

**Comment by ChristophWurst** [comment]
> Reviewers: please help me with it

Could you tell us what to test exactly?

File: apps/files_sharing/js/files_drop.js [hunk 1]
```diff
@@ -65,15 +65,15 @@
 			$('#drop-upload-progress-indicator').removeClass('hidden');
 
 			$('#drop-uploaded-files').append(output({isUploading: true, name: data.files[0].name}));
-			$('[data-toggle="tooltip"]').tooltip();
+			$('[data-toggle="tooltip"]')
```
Comment (ChristophWurst) [inline apps/files_sharing/js/files_drop.js:68]: drop this line

File: apps/files_sharing/js/files_drop.js [hunk 2]
```diff
@@ -65,15 +65,15 @@
...
 			data.submit();
 
 			return true;
 		},
 
 		updateFileItem: function (fileName, fileItem) {
 			$('#drop-uploaded-files li[data-name="' + fileName + '"]').replaceWith(fileItem);
-			$('[data-toggle="tooltip"]').tooltip();
+			$('[data-toggle="tooltip"]')
```
Comment (ChristophWurst) [inline apps/files_sharing/js/files_drop.js:76]: same

File: apps/files_sharing/src/share.js [hunk 3]
```diff
@@ -418,7 +417,7 @@ import { getCapabilities } from '@nextcloud/capabilities'
 				tooltip += '@' + server.replace(protocol, '') + serverPath
 			}
 
-			var html = '<span class="remoteAddress" title="' + escapeHTML(tooltip) + '">'
+			var html = '<span class="remoteAddress" title="' + text(tooltip) + '">'
```
Comment (ChristophWurst) [inline apps/files_sharing/src/share.js:420]: where does `text` come from and what does it do?

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: apps/files_sharing/src/share.js (hunk 3 above)
Comment (JuliaKirschenheuter) [reply apps/files_sharing/src/share.js:420]: was a misconception. reverting

**Review by PVince81 — APPROVED** [review]
:+1:

File: apps/files_sharing/js/files_drop.js [hunk 4]
```diff
@@ -65,15 +65,13 @@
...
-			$('[data-toggle="tooltip"]').tooltip();
 			data.submit();
 
 			return true;
 		},
 
 		updateFileItem: function (fileName, fileItem) {
 			$('#drop-uploaded-files li[data-name="' + fileName + '"]').replaceWith(fileItem);
-			$('[data-toggle="tooltip"]').tooltip();
 		},
```
Comment (ShGKme) [inline apps/files_sharing/js/files_drop.js:75]: Maybe also remove this attribute from HTML templates? It is not used anymore.
https://github.com/nextcloud/server/blob/master/apps/files_sharing/js/templates/files_drop.handlebars#L1

**Review by ShGKme — COMMENTED** [review]
It looks good to me.
I have checked that `.tooltip()` method is not used anywhere else.

It doesn't brakes anything, but maybe also remove `data-toggle="tooltip"` attribute from HTML templates. This attr is not used now.

File: apps/files_sharing/js/files_drop.js (hunk 4 above)
Comment (JuliaKirschenheuter) [reply apps/files_sharing/js/files_drop.js:75]: done, thank you!

**Review by ShGKme — APPROVED** [review]
👍

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
@JuliaKirschenheuter in this specific case, please recompile by running `./build/compile-handlebars-templates.sh`
it's a rare case because we usually don't touch the legacy handlebars templates

**Pull request merged**